test/evaluate.js

```diff
@@ -209,6 +237,106 @@ describe('evaluate', () => {
+        setScoresCalls.push({ participantAddresses, scores })
+        return { hash: '0x345' }
+      }
+    }
+    const logger = { log: debug, error: debug }
+    const fetchRoundDetails = () => ({ retrievalTasks: [VALID_TASK] })
+    await evaluate({
+      rounds,
+      roundIndex: 0,
+      ieContractWithSigner,
+      recordTelemetry,
+      fetchRoundDetails,
+      logger
+    })
+
+    let point = telemetry.find(p => p.name === 'retrieval_stats')
+    assert(!!point,
+      `No telemetry point "retrieval_stats" was recorded. Actual points: ${JSON.stringify(telemetry.map(p => p.name))}`)
+    debug(point.name, point.fields)
+
+    assertPointFieldValue(point, 'unique_tasks', '1i')
+    assertPointFieldValue(point, 'success_rate', '1')
+    assertPointFieldValue(point, 'participants', '1i')
+    assertPointFieldValue(point, 'inet_groups', '1i')
+    assertPointFieldValue(point, 'measurements', '1i')
+    assertPointFieldValue(point, 'download_bandwidth', '1024i')
+
+    assertPointFieldValue(point, 'result_rate_OK', '1')
+    assertPointFieldValue(point, 'result_rate_TIMEOUT', '0')
+
+    assertPointFieldValue(point, 'ttfb_p10', '1000i')
+    assertPointFieldValue(point, 'ttfb_mean', '1000i')
+    assertPointFieldValue(point, 'ttfb_p90', '1000i')
+
+    assertPointFieldValue(point, 'duration_p10', '2000i')
+    assertPointFieldValue(point, 'duration_mean', '2000i')
+    assertPointFieldValue(point, 'duration_p90', '2000i')
+
+    assertPointFieldValue(point, 'car_size_p10', '1024i')
+    assertPointFieldValue(point, 'car_size_mean', '1024i')
+    assertPointFieldValue(point, 'car_size_p90', '1024i')
+
+    point = telemetry.find(p => p.name === 'retrieval_stats_all')
+    assert(!!point,
+      `No telemetry point "retrieval_stats_all" was recorded. Actual points: ${JSON.stringify(telemetry.map(p => p.name))}`)
+    debug(point.name, point.fields)
+
+    assertPointFieldValue(point, 'unique_tasks', '2i')
+    assertPointFieldValue(point, 'success_rate', '0.5')
+    assertPointFieldValue(point, 'participants', '2i')
+    assertPointFieldValue(point, 'inet_groups', '2i')
+    assertPointFieldValue(point, 'measurements', '2i')
+    assertPointFieldValue(point, 'download_bandwidth', '3072i')
+
+    assertPointFieldValue(point, 'result_rate_OK', '0.5')
+    assertPointFieldValue(point, 'result_rate_TIMEOUT', '0')
+    assertPointFieldValue(point, 'result_rate_ERROR_500', '0.5')
+
+    assertPointFieldValue(point, 'ttfb_min', '1000i')
+    assertPointFieldValue(point, 'ttfb_mean', '5500i')
+    assertPointFieldValue(point, 'ttfb_p90', '10000i')
+
+    assertPointFieldValue(point, 'duration_p10', '2000i')
+    assertPointFieldValue(point, 'duration_mean', '11000i')
+    assertPointFieldValue(point, 'duration_p90', '20000i')
+
+    assertPointFieldValue(point, 'car_size_p10', '1024i')
+    assertPointFieldValue(point, 'car_size_mean', '1536i')
+    assertPointFieldValue(point, 'car_size_p90', '2048i')
+  })
```

Comment:
it might be beneficial to test `reportRetrievalStats` in isolation, since its big enough to eventually (I think) need its own set of test cases

Reply:
Great suggestion. I am also going to extract the implementation to a standalone file.